app/helpers/physical_server_helper/textual_summary.rb

```diff
@@ -163,6 +155,14 @@ def textual_health_state
     {:label => _("Health State"), :value => @record.health_state}
   end
 
+  def textual_network_devices
+    device = {:label => _("Network Devices"), :value => @record.hardware.nics.count, :icon => "ff ff-network-card"}
+    unless @record.hardware.nics.nil?
+      device.update(:link => url_for(:controller => 'guest_device', :action => 'show_list'))
+    end
+    device
+  end
+
```

Comment:
You will need a nested display here, as shown below -
(please make adjustments to the below method as necessary)

```rb
def textual_network_devices
  hardware_nics = @record.hardware.nics.count
  h = {:label => _("Network Devices"), :icon => "ff ff-network-card", :value => hardware_nics}
  if hardware_nics > 0
    h[:link] = "/physical_server/#{@record.id}?display=guest_devices"
  end
  h
end
```

Reply:
Note that `"/physical_server/#{@record.id}?display=guest_devices"` will work if there is an association between a physical server and a guest device.
Is it possible to add that association in the model if it does not already exist?

Reply:
@AparnaKarve I created an association between physical server and guest device, but `"/physical_server/#{@record.id}?display=guest_devices"` doesn't seem to be working. When I click on the link, I get a page with an error that says "The page you were looking for doesn't exist" and in development.log, I get these errors:
```
[----] I, [2018-02-14T06:43:08.764242 #6493:2abc084e2ea8]  INFO -- : Started GET "/physical_server/2?display=guest_devices" for 127.0.0.1 at 2018-02-14 06:43:08 -0500
[----] F, [2018-02-14T06:43:08.790984 #6493:2abc084e2ea8] FATAL -- :   
[----] F, [2018-02-14T06:43:08.791066 #6493:2abc084e2ea8] FATAL -- : ActionController::RoutingError (No route matches [GET] "/physical_server/2"):
[----] F, [2018-02-14T06:43:08.791093 #6493:2abc084e2ea8] FATAL -- :   
[----] F, [2018-02-14T06:43:08.791116 #6493:2abc084e2ea8] FATAL -- : actionpack (5.0.6) lib/action_dispatch/middleware/debug_exceptions.rb:53:in `call'
actionpack (5.0.6) lib/action_dispatch/middleware/show_exceptions.rb:31:in `call'
railties (5.0.6) lib/rails/rack/logger.rb:36:in `call_app'
railties (5.0.6) lib/rails/rack/logger.rb:26:in `call'
sprockets-rails (3.2.1) lib/sprockets/rails/quiet_assets.rb:13:in `call'
actionpack (5.0.6) lib/action_dispatch/middleware/request_id.rb:24:in `call'
rack (2.0.4) lib/rack/method_override.rb:22:in `call'
rack (2.0.4) lib/rack/runtime.rb:22:in `call'
activesupport (5.0.6) lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
actionpack (5.0.6) lib/action_dispatch/middleware/executor.rb:12:in `call'
actionpack (5.0.6) lib/action_dispatch/middleware/static.rb:136:in `call'
rack (2.0.4) lib/rack/sendfile.rb:111:in `call'
railties (5.0.6) lib/rails/engine.rb:522:in `call'
puma (3.7.1) lib/puma/configuration.rb:232:in `call'
puma (3.7.1) lib/puma/server.rb:578:in `handle_request'
puma (3.7.1) lib/puma/server.rb:415:in `process_client'
puma (3.7.1) lib/puma/server.rb:275:in `block in run'
puma (3.7.1) lib/puma/thread_pool.rb:120:in `block in spawn_thread'
```

Reply:
@AparnaKarve Is there another change that I need to make somewhere to get this to work? Thanks

Reply:
@skovic Looks like I missed a `show` action in the above url
It should have been -

```rb
"/physical_server/show/#{@record.id}?display=guest_devices"
```

Reply:
@AparnaKarve Thanks. I made that change and got a little further. Now I'm getting the following error: 
`undefined method 'display_methods' for PhysicalServerController:Class [physical_server/show] `

Reply:
I think you are just missing this method in the physical server controller -

```rb
def self.display_methods
   %w(guest_devices)
end
```

Reply:
@AparnaKarve Thanks. The page loads now, but it doesn't load a table with guest device details -- it only has the heading. I think I have the guest device / physical server relation set up correctly; I am able to access _@record.guest_devices_ from the physical server _textual_summary.rb_ file -- it returns a list of guest devices associated with the server. What are some other things I could check? Please see the screenshot below:

![image](https://user-images.githubusercontent.com/23690141/36318473-f532896c-130d-11e8-80ee-7c7df439af43.png)

Reply:
> I am able to access @record.guest_devices from the physical server textual_summary.rb file -- it returns a list of guest devices associated with the server.

Ok, that is good.

So now you need the actual `display` method in your Physical Server controller -
```rb
def display_guest_devices
  nested_list(GuestDevice, {:named_scope => :with_ethernet_type})
end
```

Note that the nested list implementation does not use `guest_device_controller` - so if it's not being used for any other use case, you can remove it

Reply:
@AparnaKarve I added the _display_guest_devices_ method, but the page still doesn't display the table of devices. It still looks like the screenshot above. There doesn't seem to be any associated errors in the logs. How can I further debug this? Thanks

Reply:
@skovic Can you push your latest code - that way I can troubleshoot it better.
Thanks.